File: csharp/bufdither_net/bufdither_net/PixelProvider.cs
using System;

namespace bufdither_net
{
	public unsafe struct RGBA
	{
		public fixed int rgba[4];
	}
	
	public unsafe interface PixelProvider
	{
		bool isInBounds (int x, int y);
    
		int getWidth ();
		int getHeight ();
		/// returns false if x,y is out of bounds
		int ofs (int x, int y);   
		void setPixelAt (int byteofs, ref RGBA rgba);
		void getPixelAt (int byteofs, ref RGBA rgba);
	}
}

File: lua/pixelprovider.lua
-- pixel interface, not used as now
PixelProvider = {
    
}

boolean isInBounds(int x, int y);
    
    int getWidth();
    int getHeight();
    /// returns false if x,y is out of bounds
    int ofs(int x, int y);   
    void setPixelAt(int byteofs, int rgba[]);
    void getPixelAt(int byteofs, int rgba[]);

function PixelProvider:w()
end

function PixelProvider:h()
end

function PixelProvider:ofs(x,y)
    return 0
end

function PixelProvider:setPixelAt(byteofs, rgba)
end

function PixelProvider:getPixelAt(byteofs, rgba)
end


